src/components/elements/input/ClientImageUploadDialog.tsx

```diff
@@ -40,7 +49,12 @@ const ClientImageUploadDialog: React.FC<ClientImageUploadDialogProps> = ({
   } = useGetClientImageQuery({
     variables: { id: clientId },
   });
-  const [mutate, { loading: updating }] = useUpdateClientImageMutation();
+  const [errors, setErrors] = useState<ErrorState>(emptyErrorState);
+  const [mutate, { loading: updating }] = useUpdateClientImageMutation({
+    onError: (apolloError) => {
```

Comment:
I think it'd be ideal style-wise for this to be moved below into the callback at line 77, but I couldn't quite get Typescript working correctly with this approach.

Or else for that callback to be moved up here into an `onCompleted` call, which better matches our general pattern -- but we need the event object for the `handleClose` object, which I think we don't have access to in the mutation-provided callbacks.

But I'd be delighted to learn that I am missing something obvious! 😅

Reply:
Yeah that was a little funky. It's doable by moving the onCompleted steps into `onCompleted` instead of a `then`, and moving handleClose up

```
  const [mutate, { loading: updating }] = useUpdateClientImageMutation({
    onCompleted: (data) => {
      const { errors: validationErrors = [] } = data.updateClientImage || {};
      if (validationErrors.length) {
        setErrors(partitionValidations(validationErrors));
      } else {
        handleClose();
      }
    },
    onError: (apolloError) => setErrors({ ...emptyErrorState, apolloError }),
  });
```

Reply:
Okay a few adjustments on https://github.com/greenriver/hmis-frontend/pull/987/commits/39a56a4939b095e0dddee46c0ade4229632170df:
* Use onCompleted and onError instead of `then(..)` pattern
* Show apollo errors from deleteImage mutation (previously swallowed)
* Stop calling refetchClient after deleteImage. This was unnecessary and caused a weird lag before the dialog get's closed. There is no need to refetch because it gets deleted from the cache right away, I tested.
* Update the `handleClose` callback so the event/reason params are optional.. this is not really needed, since in some cases we are closing it because of some logic rather than an event.